peregrine/iqgen/iqgen_main.py

```diff
@@ -0,0 +1,460 @@
+  parser.add_argument('--doppler-speed',
+                      type=float,
+                      default=-10.,
+                      help="Doppler shift change in hertz/second",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-distance',
+                      type=float,
+                      help="Distance in meters (initial)",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-amplitude',
+                      type=float,
+                      default=-10.,
+                      help="Doppler change amplitude (hertz)",
+                      action=UpdateDopplerType)
+  parser.add_argument('--doppler-period',
+                      type=float,
+                      help="Doppler change period (seconds)",
+                      action=UpdateDopplerType)
+  parser.add_argument('--amplitude-type',
+                      default="poly",
+                      choices=["poly", "sine"],
+                      help="Configure amplitude type",
+                      action=UpdateAmplitudeType)
+  parser.add_argument('--amplitude-value0',
+                      type=float,
+                      help="Amplitude coefficient A0",
+                      action=UpdateAmplitudeType)
+  parser.add_argument('--amplitude-value1',
+                      type=float,
+                      help="Amplitude coefficient A1",
+                      action=UpdateAmplitudeType)
+  parser.add_argument('--amplitude-value2',
+                      type=float,
+                      help="Amplitude coefficient A3",
+                      action=UpdateAmplitudeType)
+  parser.add_argument('--amplitude-period',
+                      type=float,
+                      help="Amplitude period",
+                      action=UpdateAmplitudeType)
+  parser.add_argument('--message-type', default="zero",
+                      choices=["zero", "one", "zero+one"],
+                      help="Message type",
+                      action=UpdateMessageType)
+  parser.add_argument('--message-file',
+                      type=argparse.FileType('rb'),
+                      help="Source file for message contents.",
+                      action=UpdateMessageFile)
+  parser.add_argument('--symbol_delay',
+                      type=int,
+                      help="Initial symbol index")
+  parser.add_argument('--chip_delay',
+                      type=int,
+                      help="Initial chip index")
+  parser.add_argument('--lpf',
+                      default=False,
+                      help="Enable low pass filter",
+                      action='store_true')
+  parser.add_argument('--snr',
+                      type=float,
+                      help="SNR for noise generation")
+  parser.add_argument('--debug',
+                      default=False,
+                      help="Enable debug output",
+                      action='store_true')
+  parser.add_argument('--interval',
```

Comment:
I would propose something like
'--generate'
help="Amount of data to generate, in seconds"